### PR TITLE
Microchip MEC15xx and MEC172x adc driver merge PR

### PR DESCRIPTION
**Pull request by Manimaran-A**
Change list:
1. "adc_mchp_xec_v2.c" driver code clean up.
2. Updated the "adc_mchp_xec_v2.c" adc driver to support both MEC172x and
MEC15xx SOC.
3. Deleted "adc_mchp_xec.c" and "microchip,xec-adc.yaml" file.
4. Renamed "adc_mchp_xec_v2.c" -> "adc_mchp_xec.c"
5. Renamed "microchip,xec-adc-v2.yaml" -> "microchip,xec-adc.yaml"
6. Added MEC15xx board overlay in sample folder.

### CODE REVIEW

**Comment by jvasanth1** [comment]
@albertofloyd can you help to review

File: dts/arm/microchip/mec1501hsz.dtsi [hunk 1]
```diff
@@ -419,9 +419,11 @@
 			compatible = "microchip,xec-adc";
 			reg = <0x40007c00 0x90>;
 			interrupts = <78 0>, <79 0>;
+			girqs = <17 8>, <17 9>;
+			pcrs = <3 3>;
 			status = "disabled";
 			#io-channel-cells = <1>;
-			clktime = <32>;
```
Comment (albertofloyd) [inline dts/arm/microchip/mec1501hsz.dtsi:424]: @Manimaran-A elaborate on this change, we observed in the past an inaccuracy on ADC readings with wrong clktime. Refer to 
https://github.com/zephyrproject-rtos/zephyr/commit/e4481f82cb02afced321adc845f771e82a541e6e

File: dts/arm/microchip/mec172xnsz.dtsi [hunk 2]
```diff
@@ -688,13 +688,14 @@
...
 		adc0: adc@40007c00 {
-			compatible = "microchip,xec-adc-v2";
+			compatible = "microchip,xec-adc";
 			reg = <0x40007c00 0x90>;
 			interrupts = <78 0>, <79 0>;
 			girqs = <17 8>, <17 9>;
 			pcrs = <3 3>;
 			status = "disabled";
 			#io-channel-cells = <1>;
+			clktime = <1>;
```
Comment (albertofloyd) [inline dts/arm/microchip/mec172xnsz.dtsi:698]: @Manimaran-A same here with clktime default value

File: dts/arm/microchip/mec1501hsz.dtsi (hunk 1 above)
Comment (albertofloyd) [reply dts/arm/microchip/mec1501hsz.dtsi:424]: cc @AdiIntel

Comment (Manimaran-A) [reply dts/arm/microchip/mec1501hsz.dtsi:424]: Before the driver merge, I observed that MEC172x driver did not explicitly initialize the config_reg. So it  uses the default value that is 1. But in MEC1501, i we configured it with 32 (through dts). 

So, when i did the MEC172x and MEC15xx merge thought of bring both as same configuration value and kept 1 in the dts.

I will update the value back to 32 for MEC1501.

Comment (AdiIntel) [reply dts/arm/microchip/mec1501hsz.dtsi:424]: clktime better be set minimum 32.

Just to share our learning, we couldn't get appropriate ADC values with MEC1501 evb card with default clktime ( = 1). Readings were off by ~30%.
Increasing this clktime value was only the solution to get better accuracy even though it adds a bit of latency.
I noticed the same issue on boards using 17xx - where readings are in-accurate, and now I know why. :) - this new driver still had default ( =1) value for clktime. Agree with you to update back to 32.

**Review by albertofloyd — CHANGES_REQUESTED** [review]
Need to change clktime back to 32 to prevent inaccurate readings as confirmed by @AdiIntel

File: dts/arm/microchip/mec1501hsz.dtsi (hunk 1 above)
Comment (Manimaran-A) [reply dts/arm/microchip/mec1501hsz.dtsi:424]: Ok, i revert the configuration back to 32

File: dts/arm/microchip/mec172xnsz.dtsi (hunk 2 above)
Comment (Manimaran-A) [reply dts/arm/microchip/mec172xnsz.dtsi:698]: Ok, i update the configuration by 32

**Review by albertofloyd — APPROVED** [review]

**Review by jvasanth1 — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**